cmd/generic-handlers.go

```diff
@@ -469,3 +475,75 @@ func (h pathValidityHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 	h.handler.ServeHTTP(w, r)
 }
+
+// logginHandler logs all HTTP requests and responses
+// and write the log to a io.Writer object.
+type loggingHandler struct {
+	writer    io.Writer
+	writeLock *sync.Mutex
+	handler   http.Handler
+}
```

Comment:
more info: https://twitter.com/davecheney/status/743289114980548612

using `sync.Mutex` instead of `*sync.Mutex` means we have to do pointer receiver:
func (h *loggingHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {